examples/python/wasm-in-pytket.py

```diff
@@ -0,0 +1,136 @@
+c1 = c.add_c_register("c1", 4)
+c2 = c.add_c_register("c2", 5)
+
+c.add_wasm_to_reg(
+    "multi",  # function name
+    wfh,  # wasm file handler
+    [c0, c1],  # register for each input parameter
+    [c2],
+)  # register for the result
+c.add_wasm_to_reg(
+    "add_one",  # function name
+    wfh,  # wasm file handler
+    [c2],  # register for each input parameter
+    [c2],
+)  # register for the result
+
+# The WASM might have some global data stored. To make sure this data is not messed up by function calls in the wrong order pytket will make sure that the order of the wasm calls within a circuit is not restructured. For this purpose pytket will add all wasm operation to a wasm_wire by default. If you are not worried about a possible restructure of the wasm calls in your circuit you have the option to not add the wasm_wire to your wasm operations. If you only want to stop some special reordering for some of your wasm operations you can add some the wasm operations to multiple wasm_wire to allow the restructuring in the intended way. Even if there are not wasm_wire given, pytket will only restructure the the wasm operations if there are no dependencies to in parameters or the results.
+
+# Here you can see that all operations we have created above are conected to the default wasm_wire:
+
+for g in c:
+    print(g)
+
+# We will now create a new circuit and add four operations. The two add_one operations should be allowed to commute, but we want to make sure that "multi" is executed after the two other functions. The last add_two operation can commute with all others.
+
+c = Circuit(0, 5)
+
+c.add_wasm("add_one", wfh, [1], [1], [Bit(0), Bit(0)], [0])
+c.add_wasm("add_one", wfh, [1], [1], [Bit(1), Bit(1)], [1])
+c.add_wasm("multi", wfh, [1, 1], [1], [Bit(2), Bit(3), Bit(2)], [0, 1])
+c.add_wasm("add_two", wfh, [1], [1], [Bit(4), Bit(4)], [])
+
+for g in c:
+    print(g)
+
+# One helpful feature might be to plot the DAG of the circuit to get an overview of the different
+
+from pytket.utils import Graph
+
+g = Graph(c)
+g.view_DAG()
+
+# ## Send WASM to the Backend
+# In the last step we want to send the circuit with the wasm to a backend. First we create the backend. For this step you will need Quantinuum credentials.
+
+from pytket.extensions.quantinuum import QuantinuumBackend
+
+machine = "H1-1E"
+b = QuantinuumBackend(device_name=machine)
+b.login()
+
+# When processing the circuit you need to add the wasmfilehandler you created as parameter to the `process_circuits` in the shown way
+
+from pytket.extensions.quantinuum import QuantinuumBackend
+
+c = Circuit(1)
+c.name = "test_wasm"
+a = c.add_c_register("a", 8)
+c.add_wasm_to_reg("add_one", wfh, [a], [a])
+c = b.get_compiled_circuit(c)
+h = b.process_circuits([c], n_shots=10, wasm_file_handler=wfh)[0]
+
+status = b.circuit_status(h)
+print(status)
+
+status = b.circuit_status(h)
+print(status)
+
+result = b.get_result(h)
+print(result)
+
+for shot in result.get_shots():
```

Comment:
Its nice to show such an example with the `QuantinuumBackend`. I was wondering if there is a short demo we could make where the WASM is used in the circuit to do something rather than just showing the API. If this comment is unclear then let me know